src/AutoBus.ts

```diff
@@ -0,0 +1,59 @@
+import assert from 'assert';
+import LocalTransport, { Listener, Transport } from './local/LocalTransport';
+
+export interface NextFunc<T> {
+  (error: Error | null, message: T): void;
```

Comment:
`message` will likely be `undefined` if `err` is not `null`, and `T` if `err` is `null`. Add overload? Try like this:

```ts
  (error: null, message: T): void;
  (error: Error, message: undefined): void;
  (error: Error | null, message?: T): void;
```